src/Symfony/Bundle/FrameworkBundle/Tests/Console/ApplicationTest.php

```diff
@@ -258,14 +258,25 @@ private function getKernel(array $bundles, $useDispatcher = false)
         $container
             ->expects($this->exactly(2))
             ->method('hasParameter')
-            ->withConsecutive(['console.command.ids'], ['console.lazy_command.ids'])
-            ->willReturnOnConsecutiveCalls(true, true)
+            ->willReturnCallback(function (string $arg) {
```

Comment:
what about this pattern?

```php
->willReturnCallback(function (...$args) {
    static $series = [
		[['console.command.ids'], true],
        [['console.lazy_command.ids'], true],
	];

    [$expectedArgs, $return] = array_shift($series);

    return $expectedArgs === $args ? $return : $this->fail();
})
```

Reply:
This is super clever, I love it. I've been trying to implement it all over PR. It's super clean.
Thanks for this solution!

Reply:
Thank you for that idea. I've built class oriented solution for that https://gist.github.com/bizley/d4bd39d0c4a1b4106c56845ff0c07068 (for simplified calling and to get readable message on error) - feel free to use if you'll find it useful.

Reply:
using assertSame, nice idea!

```php
->willReturnCallback(function (...$args) {
    static $series = [
        [['console.command.ids'], true],
        [['console.lazy_command.ids'], true],
    ];

    [$expectedArgs, $return] = array_shift($series);
    $this->assertSame($expectedArgs, $args);

    return $return;
})

Reply:
Alright, I updated everything! Also, this pattern allowed me to address the remaining `withConsecutive()`. So this PR bundles changes all across the codebase and there should be no call to this method anymore 🙂